src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -152,10 +162,28 @@ export class UiSettingsClient implements IUiSettingsClient {
       // default will get from all scope and merge
       // loop UiSettingScopeReadOptions
       for (const readOptions of UiSettingScopeReadOptions) {
-        userProvided = { ...userProvided, ...this.onReadHook<T>(await this.read(readOptions)) };
+        if (readOptions.scope !== UiSettingScope.WORKSPACE) {
```

Comment:
I do not think there will be difference between read user level setting and workspace level setting.

What about
- read config with scope
- manually set the value as undefined if the key can not be found from the response
- trim keys that is not in the scope
- loop the steps above.